xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-api/src/main/java/org/xwiki/security/authentication/UserAuthenticatedEvent.java

```diff
@@ -78,9 +67,21 @@ public UserReference getUserReference()
 
     @Override
     public boolean matches(Object other)
+    {
+        return other instanceof UserAuthenticatedEvent;
+    }
+
+    @Override
+    public boolean equals(Object other)
```

Comment:
My point in the previous comment was that the implementation of matches was wrong and not that you should implement equals (it's useless IMO). Also, your hashCode implementation was wrong (it's supposed to be consistent with equals) but anyway you should remove both equals and hashCode since we don't really need them.

Reply:
I got it. It will be fixed.